araproc/analysis/standard_reco.py

```diff
@@ -582,34 +583,22 @@ def get_surface_corr_max(self, corr_map, z_thresh=-10):
         if hist is None:
             raise ValueError("The 'map' key was not found in corr_map.")
         
-        surf_corr_values = []
-        corr_bins = []
-
-        # Loop through the bins and store values within theta range
-        for x_bin in range(1, hist.GetNbinsX() + 1):
-            for y_bin in range(1, hist.GetNbinsY() + 1):
-                theta = hist.GetYaxis().GetBinCenter(y_bin)
-                phi = hist.GetXaxis().GetBinCenter(x_bin)
-                corr = hist.GetBinContent(x_bin, y_bin)
-                if theta_min <= theta <= theta_max:
-                    surf_corr_values.append(corr)
-                    corr_bins.append((x_bin, y_bin))
-
-        # Identify maximum surface correlation
-        if not surf_corr_values:
-            raise RuntimeError("No correlation values found in the specified theta range.")
+        # set the range of the histogram to only be where it's relevant
+        theta_min_bin = hist.GetYaxis().FindBin(theta_min)
+        theta_max_bin = hist.GetYaxis().FindBin(theta_max)
+        hist.GetYaxis().SetRange(theta_min_bin+1, theta_max_bin-1)
```

Comment:
Why the +1 and -1? I wonder if this would be more straight forward with `hist.GetYaxis().SetRangeUser(theta_min, theta_max)`? But maybe the approximations it makes are a problem.

Reply:
Neat trick btw!

Reply:
I think it's the approximations that are biting me here. Somehow using the SetRange functions (either in bin space or user space) doesn't precisely replicate the for loop, and I'm not sure why. The +1 and -1 was the only way I got this to replicate the for loop exactly.

Reply:
I'm not sure what to do about it. Any thoughts?